src/traces/bar/plot.js

```diff
@@ -215,14 +214,10 @@ function plot(gd, plotinfo, cdModule, traceLayer, opts, makeOnCompleteCallback)
                 var op = Color.opacity(mc);
                 var fixpx = (op < 1 || lw > 0.01) ? roundWithLine : expandToVisible;
 
-                if(spansHorizontal) {
-                    x0 = fixpx(x0, x1);
-                    x1 = fixpx(x1, x0);
-                }
-                if(spansVertical) {
-                    y0 = fixpx(y0, y1);
-                    y1 = fixpx(y1, y0);
-                }
+                x0 = fixpx(x0, x1, isHorizontal);
+                x1 = fixpx(x1, x0, isHorizontal);
+                y0 = fixpx(y0, y1, !isHorizontal);
+                y1 = fixpx(y1, y0, !isHorizontal);
```

Comment:
Can you explain how this commit changes the logic.

Reply:
All four positions i.e. x0, x1, y0, y1 (not only two of hem) should possibly be rounded by `fixpx` function.
In case the bar is `horizontal` the `fixpx` function may decide not to round (floor and ciel) zero spans i.e. when `x0 === x1`.
Similarly, when a bar is `vertical` the `fixpx` will return early for zero spans i.e. when `y0 === y1`.
However, in case of having narrow lines e.g. a `vertical` bar the `fixpx` should round values to upper and lower values even if when `x0 === x1`. Also when a `horizontal` bar encounters `y0 === y1` narrow lines should be displayed simalr to v1.51.1 and earlier versions.

Reply:
Ok, can you try to write a comment explaining that above this line:

```js
if(hideZeroSpan && v === vc) return v;
```

??

Reply:
Done in https://github.com/plotly/plotly.js/pull/4568/commits/c1ee8c2279058d61e7075a9967899a038b75e45e.